Cargo.lock

```diff
@@ -562,7 +579,7 @@ dependencies = [
 [[package]]
 name = "binary-merkle-tree"
 version = "4.0.0-dev"
-source = "git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.41#3bd809d59c31c7205a0f36c6ddcba603e43051fc"
+source = "git+https://github.com/paritytech/substrate?branch=polkadot-v0.9.42#3bb3882ce094ac211dea93fabfdcd4f902f5a7fd"
```

Comment:
You don't seem to be up to date with `polkadot-v0.9.42`. Try running `cargo update`. This will update all dependencies. You could run `cargo update -p sp-io` which updates most of the substrate dependencies. Make sure to also check cumulus & polkadot repositories.